mvhashmap_test.go

```diff
@@ -2,26 +2,182 @@ package block_stm
 
 import (
 	"fmt"
-	"github.com/stretchr/testify/require"
+	"math/rand"
 	"testing"
+
+	"github.com/stretchr/testify/require"
 )
 
+// create data for a given txIdx and incarnation
 func valueFor(txIdx, inc int) []byte {
 	return []byte(fmt.Sprintf("%ver:%ver:%ver", txIdx*5, txIdx+inc, inc*5))
 }
 
+// Result of the following benchmark test:
+// goos: darwin
+// goarch: arm64
+// pkg: github.com/paulgoleary/go-block-stm
+// BenchmarkWriteTimeSameLocationDifferentTxIdx-8   	  994430	      1242 ns/op	     256 B/op	      10 allocs/op
+// PASS
+// ok  	github.com/paulgoleary/go-block-stm	2.636s
+func BenchmarkWriteTimeSameLocationDifferentTxIdx(b *testing.B) {
+	mvh2 := MakeMVHashMap()
+	ap2 := []byte("/foo/b")
+
+	randInts := []int{}
+	for i := 0; i < b.N; i++ {
+		randInts = append(randInts, rand.Intn(1000000000000000))
+	}
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		mvh2.Write(ap2, Version{randInts[i], 1}, valueFor(randInts[i], 1))
+	}
+}
+
+// Result of the following benchmark test:
+// goos: darwin
+// goarch: arm64
+// pkg: github.com/paulgoleary/go-block-stm
+// BenchmarkReadTimeSameLocationDifferentTxIdx-8   	 1388722	       965.2 ns/op	      24 B/op	       3 allocs/op
+// PASS
+// ok  	github.com/paulgoleary/go-block-stm	5.658s
+func BenchmarkReadTimeSameLocationDifferentTxIdx(b *testing.B) {
+	mvh2 := MakeMVHashMap()
+	ap2 := []byte("/foo/b")
+	txIdxSlice := []int{}
+	for i := 0; i < b.N; i++ {
+		txIdx := rand.Intn(1000000000000000)
+		txIdxSlice = append(txIdxSlice, txIdx)
+		mvh2.Write(ap2, Version{txIdx, 1}, valueFor(txIdx, 1))
```

Comment:
Should include mvh2.Write in the benchmark right?

Reply:
`BenchmarkReadTimeSameLocationDifferentTxIdx` is only benchmarking the read time. If you would like to have a separate benchmark which includes write as well as read, I can do that as well.

Reply:
Okay I see. I am fine with it. Thanks for the explanation!